pyccel/codegen/printing/ccode.py

```diff
@@ -1233,14 +1234,12 @@ def _print_NumpyUfuncBase(self, expr):
             numpy.cos(x) ==> cos(x)
 
         """
-        # add necessary include
+        self.add_import(c_imports['math'])
         type_name = type(expr).__name__
         try:
             func_name = numpy_ufunc_to_c_float[type_name]
-            if func_name in ('sign', 'csign'):
-                self.add_import(c_imports['ndarrays'])
-            else:
-                self.add_import(c_imports['math'])
+            if func_name in ('sign'):
```

Comment:
I don't think this will work. `('sign')` is not a tuple, it is a string. You are checking if the function name is one of `s`, `i`, `g`, or `n`. You should use `if func_name in ('sign',):`, or (more logically when only comparing with 1 value `if func_name == 'sign':`

Reply:
True! The fact that all tests pass suggests that we need much stricter unit testing.

Reply:
After a quick check it seems that `func_name in 'sign'` is not equivalent to `func_name in ('s','i','g','n')` so I don't think this problem can cause failing tests. It can just introduce unnecessary imports
```
In [3]: func_name in 'sign'                                                                                                                                                                                
Out[3]: True

In [4]: 's' in 'sign'                                                                                                                                                                                      
Out[4]: True
```

Reply:
All these lines have been removed, as they are no longer needed.